examples/src/chains/retrieval_qa_custom.ts

```diff
@@ -1,32 +1,86 @@
 import { OpenAI } from "langchain/llms/openai";
-import { RetrievalQAChain, loadQAMapReduceChain } from "langchain/chains";
 import { HNSWLib } from "langchain/vectorstores/hnswlib";
 import { OpenAIEmbeddings } from "langchain/embeddings/openai";
 import { RecursiveCharacterTextSplitter } from "langchain/text_splitter";
 import * as fs from "fs";
+import { PromptTemplate } from "langchain/prompts";
+import {
+  RunnablePassthrough,
+  RunnableSequence,
+} from "langchain/schema/runnable";
+import { StringOutputParser } from "langchain/schema/output_parser";
 
 // Initialize the LLM to use to answer the question.
 const model = new OpenAI({});
-const text = fs.readFileSync("state_of_the_union.txt", "utf8");
+const text = fs.readFileSync("examples/state_of_the_union.txt", "utf8");
 const textSplitter = new RecursiveCharacterTextSplitter({ chunkSize: 1000 });
 const docs = await textSplitter.createDocuments([text]);
 
-// Create a vector store from the documents.
+// Default prompt used in the loadQAMapReduceChain for getting more fine grained context
+// from returned relevant docs.
+const qaTemplate =
+  PromptTemplate.fromTemplate(`Use the following portion of a long document to see if any of the text is relevant to answer the question. 
+Return any relevant text verbatim.
+{context}
+Question: {question}
+Relevant text, if any:`);
+
+// Create a new chain that uses the default prompt and the relevant docs.
+const queryDocsChain = qaTemplate.pipe(model).pipe(new StringOutputParser());
+
+// Default prompt used in the loadQAMapReduceChain for getting a final answer.
+const combineDocsPrompt = PromptTemplate.fromTemplate(
+  `Given the following extracted parts of a long document and a question, create a final answer. 
+If you don't know the answer, just say that you don't know. Don't try to make up an answer.
+
+QUESTION: {question}
+=========
+SUMMARIES: {summaries}
+=========
+FINAL ANSWER:`
+);
+
+// Create a new chain using the default prompt and an output parser.
+const combineDocsChain = combineDocsPrompt
+  .pipe(model)
+  .pipe(new StringOutputParser());
+
+const query = "What did the president say about Justice Breyer?";
+
+// Create a vector store retriever from the documents.
 const vectorStore = await HNSWLib.fromDocuments(docs, new OpenAIEmbeddings());
+const retriever = vectorStore.asRetriever();
+
+const relevantDocs = await retriever.getRelevantDocuments(query);
+
+const serializedDocsContent = relevantDocs.map((doc) => doc.pageContent);
+
+// Perform a map over all the relevant docs and query the LLM.
+const queryDocsChainResults = await Promise.all(
```

Comment:
Use a `mapReduce` chain here directly as discussed

Reply:
Pushed up a commit updating it to use `loadQAMapReduceChain`, however I couldn't see where I could also use Runnables. Could this just be an instance where Runnables aren't ready for that type of implementation?